float32 current_speed_l
float32 current_speed_r